File: backend/src/main/java/com/mentorplatform/model/Session.java
package com.mentorplatform.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "sessions")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Session {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "mentor_id", nullable = false)
    private User mentor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "student_id")
    private User student;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    @Builder.Default
    private String language = "javascript";

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private Status status = Status.PENDING;

    @Column(name = "created_at", nullable = false)
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    @Column(name = "started_at")
    private LocalDateTime startedAt;

    @Column(name = "ended_at")
    private LocalDateTime endedAt;

    public enum Status { PENDING, ACTIVE, ENDED }
}

File: frontend/src/types/session.ts
export type SessionStatus = 'PENDING' | 'ACTIVE' | 'ENDED';

export interface Session {
  id: string;
  mentorId: number;
  studentId: number | null;
  mentorName: string;
  studentName: string | null;
  status: SessionStatus;
  title: string;
  language: string;
  createdAt: string;
  startedAt: string | null;
  endedAt: string | null;
}

export interface CreateSessionRequest {
  title: string;
  language: string;
}

export interface CodeUpdate {
  sessionId: string;
  code: string;
  language: string;
  senderId: number;
}
